### ISSUE TITLE
Lots of "Catalog construction errors" in Microsoft.VisualStudio.Default.err

### ISSUE THREAD

**Issue by jcansdale** [github/VisualStudio#954]
- GitHub Extension for Visual Studio  version: 2.2.0.8
- Visual Studio version: VS 2015/2017

There are lots of exceptions in the error log file (7 pages!) from when Visual Studio constructs its MEF catalog. Here is the `Microsoft.VisualStudio.Default.err` log from VS 2017:
[Microsoft.VisualStudio.Default.err.txt](https://github.com/github/VisualStudio/files/927920/Microsoft.VisualStudio.Default.err.txt)

This is a known issue (and workaround) caused by a version change to the `Microsoft.TeamFoundation.Controls` assembly between VS 2015 and 2017.

The `GitHub.TeamFoundation.14` assembly is compiled against `Microsoft.TeamFoundation.Controls, Version=14.0.0.0` and intended to rune inside VS 2015. The `GitHub.TeamFoundation.15` assembly is compiled against `Microsoft.TeamFoundation.Controls, Version=15.0.0.0` and intended to rune inside VS 2017.

Unfortunately it isn't possible to specify different assemblies for different versions of Visual Studio inside a VSIX file, so Visual Studio attempts to install both assemblies which results in the errors.

*Update: is is possible in recent versions of Visual Studio 2017 See* https://github.com/github/VisualStudio/commit/c9703cd2a284840474b191449c52bb67b181a5e4

I wonder if it's possible to somehow forward MEF exports to the `GitHub.TeamFoundation.14` and `GitHub.TeamFoundation.15` assemblies and avoid these errors?

**Comment by jcansdale** [comment]
Here is a dump showing where the parts inside `GitHub.TeamFoundation.15` are being consumed.

```
GitHub.ViewModels.NotAGitHubRepositoryViewModel
   .(teamExplorerServices = GitHub.Services.TeamExplorerServices)
GitHub.ViewModels.PullRequestCreationViewModel
   .(teservice = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.ViewModels.PullRequestDetailViewModel
   .(teservice = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.ViewModels.PullRequestListViewModel
   .(teservice = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.Services.RepositoryCloneService
   .(vsGitServices = GitHub.Services.VSGitServices)
GitHub.Services.RepositoryPublishService
   .(vsGitServices = GitHub.Services.VSGitServices)
GitHub.VisualStudio.ConnectionManager
   .(vsGitServices = GitHub.Services.VSGitServices)
GitHub.VisualStudio.UI.Views.GitHubPaneViewModel
   .(holder = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.VisualStudio.UI.Views.PullRequestDetailView
   .TeamExplorerServiceHolder = GitHub.VisualStudio.Base.TeamExplorerServiceHolder
GitHub.VisualStudio.UI.Views.Controls.GistCreationControl
   .(teServices = GitHub.Services.TeamExplorerServices)
GitHub.VisualStudio.UI.Views.Controls.RepositoryPublishControl
   .(teServices = GitHub.Services.TeamExplorerServices)
GitHub.VisualStudio.TeamExplorer.Home.GitHubHomeSection
   .(holder = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.VisualStudio.TeamExplorer.Home.GraphsNavigationItem
   .(holder = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.VisualStudio.TeamExplorer.Home.IssuesNavigationItem
   .(holder = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.VisualStudio.TeamExplorer.Home.PullRequestsNavigationItem
   .(holder = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.VisualStudio.TeamExplorer.Home.PulseNavigationItem
   .(holder = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.VisualStudio.TeamExplorer.Home.WikiNavigationItem
   .(holder = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.VisualStudio.TeamExplorer.Connect.GitHubConnectSection0
   .(holder = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.VisualStudio.TeamExplorer.Connect.GitHubConnectSection1
   .(holder = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
GitHub.VisualStudio.TeamExplorer.Sync.GitHubPublishSection
   .(holder = GitHub.VisualStudio.Base.TeamExplorerServiceHolder)
Microsoft.VisualStudio.TeamFoundation.TeamExplorer.Home.ProjectSection
   .NavigationItems = GitHub.VisualStudio.TeamExplorer.Home.GraphsNavigationItem
   .NavigationItems = GitHub.VisualStudio.TeamExplorer.Home.IssuesNavigationItem
   .NavigationItems = GitHub.VisualStudio.TeamExplorer.Home.PullRequestsNavigationItem
   .NavigationItems = GitHub.VisualStudio.TeamExplorer.Home.PulseNavigationItem
   .NavigationItems = GitHub.VisualStudio.TeamExplorer.Home.WikiNavigationItem
Microsoft.VisualStudio.TeamFoundation.TeamExplorer.ConnectPage.InvitationSection
   .InvitationFactories = GitHub.VisualStudio.TeamExplorer.Connect.GitHubInvitationSection

Involved assemblie:
GitHub.App.dll
GitHub.VisualStudio.dll
GitHub.TeamFoundation.15.dll
Microsoft.VisualStudio.TeamFoundation.dll
```

**Comment by jcansdale** [comment]
Fixed in #1705 / https://github.com/github/VisualStudio/commit/c9703cd2a284840474b191449c52bb67b181a5e4
